src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,50 +1,151 @@
 package core.basesyntax;
 
 import java.util.List;
+import java.util.Objects;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    public MyLinkedList() {
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == 0) {
+            head = new Node<>(null, value, null);
+            tail = head;
+            size++;
+            return;
+        }
+        Node<T> toAdd = new Node<>(null, value, null);
+        tail.next = toAdd;
+        toAdd.prev = tail;
+        tail = toAdd;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException("Wrong index " + index);
+        }
+        if (index == size) {
+            add(value);
+            return;
+        }
+        Node<T> previous = findNode(index).prev;
+        Node<T> next = findNode(index);
+        Node<T> current = new Node<>(previous, value, next);
+        if (previous == null) {
+            head = current;
+        } else {
+            previous.next = current;
+        }
+        if (next == null) {
+            tail = current;
+        } else {
+            next.prev = current;
+        }
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (T t : list) {
```

Comment:
`t` isn't informative name of the variable

Reply:
Fixed.